presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -138,9 +161,11 @@ public ResourceGroupInfo getInfo()
                     new DataSize(softMemoryLimitBytes, BYTE),
                     maxRunningQueries,
                     maxQueuedQueries,
-                    runningQueries.size() + descendantRunningQueries,
-                    queuedQueries.size() + descendantQueuedQueries,
+                    canRunMore() ? CAN_RUN : (canQueueMore() ? CAN_QUEUE : BLOCKED),
+                    eligibleSubGroups.size(),
                     new DataSize(cachedMemoryUsageBytes, BYTE),
+                    infos.stream().flatMap(info -> info.getAggregatedRunningQueries().stream()).collect(Collectors.toList()),
```

Comment:
We only want to show the path from user's resource group to root. If user is blocked on their parent resource group, we want to show what are the queries running on that resource group, which should include running queries from all the sibling groups. This is one way to achieve that. We only call this function on root so we only do this aggregation once. I'd be happy to reimplement it in a more efficient way if there are better alternatives.